plugins/docker/docker_

```diff
@@ -319,155 +509,22 @@ def main():
-            print("graph_info This graph shows docker container CPU usage.")
-            print("graph_total Total CPU usage")
-            for container in client.all_containers:
-                print("{}.label {}".format(container.name, container.name))
-                print("{}.draw AREASTACK".format(container.name))
-                print("{}.info {}".format(container.name, container_attributes(container)))
-        else:
-            print_containers_cpu(client)
-    elif wildcard == "memory":
-        if mode == "config":
-            print("graph_title Docker containers memory usage")
-            print("graph_args --base 1024 -l 0")
-            print("graph_vlabel Bytes")
-            print("graph_category virtualization")
-            print("graph_info This graph shows docker container memory usage.")
-            print("graph_total Total memory usage")
-            for container in client.all_containers:
-                print("{}.label {}".format(container.name, container.name))
-                print("{}.draw AREASTACK".format(container.name))
-                print("{}.info {}".format(container.name, container_attributes(container)))
-        else:
-            print_containers_memory(client)
-    elif wildcard == "network":
-        if mode == "config":
-            print("graph_title Docker containers network usage")
-            print("graph_args --base 1024 -l 0")
-            print("graph_vlabel bits in (-) / out (+) per ${graph_period}")
-            print("graph_category virtualization")
-            print("graph_info This graph shows docker container network usage.")
-            print("graph_total Total network usage")
-            for container in client.all_containers:
-                print("{}_down.label {}_received".format(container.name, container.name))
-                print("{}_down.type DERIVE".format(container.name))
-                print("{}_down.min 0".format(container.name))
-                print("{}_down.graph no".format(container.name))
-                print("{}_down.cdef {}_down,8,*".format(container.name, container.name))
-                print("{}_up.label {}".format(container.name, container.name))
-                print("{}_up.draw LINESTACK1".format(container.name))
-                print("{}_up.type DERIVE".format(container.name))
-                print("{}_up.min 0".format(container.name))
-                print("{}_up.negative {}_down".format(container.name, container.name))
-                print("{}_up.cdef {}_up,8,*".format(container.name, container.name))
-                print("{}_up.info {}".format(container.name, container_attributes(container)))
-        else:
-            print_containers_network(client)
+    if wildcard in series:
+        # dereference the function name by looking in the globals()
+        globals()[wildcard](client, mode)
+    elif wildcard == 'multi':
+        for s in series:
+            print(f'multigraph docker_{s}')
+            globals()[s](client, mode)
```

Comment:
I think, specifying a dict (mapping the requested features to their corresponding functions) would be a bit less hacky :)

Reply:
Heh, I was hoping to avoid this, the dict is completely redundant code.